// generate the skeleton and stub classes
// generate calculator_pb2.py and calculator_pb2_grpc.py.
// python -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. calculator.proto

syntax = "proto3";

package cloud;

service UserService {
  rpc login (Request) returns (Response);
  rpc get_status (Request) returns (Response); // For quota and simulation status
  rpc upload_file (Request) returns (Response); // For initiating and completing a virtual upload
}

message Request {
  string login = 1;
  string password = 2; // only used for login
  string file_name = 3; // NEW FIELD for file uploads
  int64 file_size = 4;   // NEW FIELD for file uploads (in bytes)
}

message Response {
  string result = 1;
}